garrysmod/gamemodes/terrortown/gamemode/corpse.lua

```diff
@@ -36,6 +36,12 @@ end
 -- If detective mode, announce when someone's body is found
 local bodyfound = CreateConVar("ttt_announce_body_found", "1")
 
+-- opportunity for addons to introduce ways for traitors(ideally) to interrupt corpse searching.
+function GM:TTTInterruptCorpseIdentify(identifier,corpse,corpse_is_traitor)

```

Comment:
I opted for default implementation as false(or nil), in case modders wanted to easily break out of all hook checks without interrupting the id/search, but have no arguments against your suggestion there.